File: SDKs/amazonaws.com/dax/2017-04-19/python/src/sdk/models/shared/updateparametergrouprequest.py
from dataclasses import dataclass, field
from typing import List
from dataclasses_json import dataclass_json
from . import parameternamevalue


@dataclass_json
@dataclass
class UpdateParameterGroupRequest:
    parameter_group_name: str = field(default=None, metadata={'dataclasses_json': { 'field_name': 'ParameterGroupName' }})
    parameter_name_values: List[parameternamevalue.ParameterNameValue] = field(default=None, metadata={'dataclasses_json': { 'field_name': 'ParameterNameValues' }})
    

File: SDKs/amazonaws.com/dax/2017-04-19/typescript/src/sdk/models/shared/updateparametergrouprequest.ts
import { Metadata, SpeakeasyBase } from "../../../internal/utils/utils";
import * as shared from "../shared";
import { ParameterNameValue } from "./parameternamevalue";


export class UpdateParameterGroupRequest extends SpeakeasyBase {
  @Metadata({ data: "json, name=ParameterGroupName" })
  parameterGroupName: string;

  @Metadata({ data: "json, name=ParameterNameValues", elemType: shared.ParameterNameValue })
  parameterNameValues: ParameterNameValue[];
}
